### PR TITLE
Unhardcode more bionics, add weapon dispersion enchantment

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
Add trait/bionic flag for thermometer and enchantment for firearm dispersion

### CODE REVIEW

File: src/ranged.cpp [hunk 1]
```diff
@@ -1777,8 +1777,11 @@ dispersion_sources Character::get_weapon_dispersion( const item &obj ) const
 
     dispersion.add_range( dispersion_from_skill( avgSkill, weapon_dispersion ) );
 
-    if( has_bionic( bio_targeting ) ) {
-        dispersion.add_multiplier( 0.75 );
+    for( const bionic &bio : *my_bionics ) {
+        cata::optional<float> dis = bio.info().dispersion_mod;
+        if( dis.has_value() ) {
+            dispersion.add_multiplier( dis.value() );
```
Comment (BrettDong) [inline src/ranged.cpp:1783]:
```
/src/ranged.cpp:1783:38: error: ‘dis’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
             dispersion.add_multiplier( dis.value() );
             ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```
Is this a compiler bug?

File: src/bionics.h [hunk 2]
```diff
@@ -79,6 +79,8 @@ struct bionic_data {
     std::map<bodypart_str_id, size_t> cut_protec;
     /**Amount of bullet protection offered by this bionic*/
     std::map<bodypart_str_id, size_t> bullet_protec;
+    /**Amount this modifies dispersion.*/
+    cata::optional<float> dispersion_mod;
```
Comment (anothersimulacrum) [inline src/bionics.h:83]: Should fix that compiler warning.
```suggestion
    cata::optional<float> dispersion_mod = cata::nullopt;
```

**Comment by SirPendrak** [comment]
Would it be possible to add some similar dispersion reducing property for bipod?

**Comment by Ramza13** [comment]
> 
> 
> Would it be possible to add some similar dispersion reducing property for bipod?

So I just updated this to make the dispersion be an enchantment instead of a bionic property.  I am not an expert on weapon mods.  I'm not sure if you can add enchantments to them or not.  If you can then yes this will work, if not that sounds like something that should be added anyway as enchanted weapon mods seems like a great idea.

**Comment by KorGgenT** [comment]
i hadn't thought of enchantments on gunmods, but if you can't do that then let me know and i'll take a crack at it after 0.F

**Pull request merged**